### PR TITLE
KAFKA-3346: Refactor & Rename Mode to ConnectionMode

### PR DESCRIPTION
**Pull request by abhi-ksolves**
I have renamed the Mode enum to ConnectionMode in the clients to improve clarity. The previous name, Mode, was undocumented and ambiguous. This made it unclear whether the connection was for an SSL client or an SSL server.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by chia7712** [comment]
@abhi-ksolves could you please share you jira account to me? I'd like to assign KAFKA-3346 to you

**Comment by abhi-ksolves** [comment]
@chia7712 I have already assigned myself to this ticket.

**Review by chia7712 — APPROVED** [review]
LGTM

File: clients/src/test/java/org/apache/kafka/common/security/ssl/SslFactoryTest.java [hunk 1]
```diff
@@ -603,7 +603,7 @@ private KeyStore sslKeyStore(Map<String, Object> sslConfig) {
         return store.get();
     }
 
-    private TestSslUtils.SslConfigsBuilder sslConfigsBuilder(Mode mode) {
-        return new TestSslUtils.SslConfigsBuilder(mode).tlsProtocol(tlsProtocol);
+    private TestSslUtils.SslConfigsBuilder sslConfigsBuilder(ConnectionMode connectionMode) {
+        return new TestSslUtils.SslConfigsBuilder(connectionMode).tlsProtocol(tlsProtocol);
     }
-}
+}
```
Comment (chia7712) [inline clients/src/test/java/org/apache/kafka/common/security/ssl/SslFactoryTest.java]: please revert this unrelated change

File: clients/src/test/java/org/apache/kafka/common/security/ssl/SslFactoryTest.java [hunk 2]
```diff
@@ -502,15 +502,15 @@ public void testEngineFactoryClosed() throws Exception {
...
                 .createNewTrustStore(trustStoreFile)
                 .useClientCert(false)
                 .build();
         serverSslConfig.put(SslConfigs.SSL_ENGINE_FACTORY_CLASS_CONFIG, TestSslUtils.TestSslEngineFactory.class);
-        SslFactory sslFactory = new SslFactory(Mode.SERVER);
+        SslFactory sslFactory = new SslFactory(ConnectionMode.SERVER);
         sslFactory.configure(serverSslConfig);
         assertInstanceOf(TestSslUtils.TestSslEngineFactory.class, sslFactory.sslEngineFactory(),
-            "SslEngineFactory must be of expected type");
+                "SslEngineFactory must be of expected type");
```
Comment (chia7712) [inline clients/src/test/java/org/apache/kafka/common/security/ssl/SslFactoryTest.java]: please revert this unrelated change

File: core/src/test/scala/unit/kafka/utils/TestUtils.scala [hunk 3]
```diff
@@ -655,7 +655,7 @@ object TestUtils extends Logging {
    * @param needsClientCert If not empty, a flag which indicates if client certificates are required. By default
    *                        client certificates are generated only if securityProtocol is SSL (not for SASL_SSL).
    */
-  def securityConfigs(mode: Mode,
+  def securityConfigs(mode: ConnectionMode,
```
Comment (chia7712) [inline core/src/test/scala/unit/kafka/utils/TestUtils.scala]: please update doc also

**Review by chia7712 — COMMENTED** [review]
@abhi-ksolves overall +1 some trivia comments PTAL

File: core/src/test/scala/unit/kafka/utils/TestUtils.scala (hunk 3 above)
Comment (abhi-ksolves) [reply core/src/test/scala/unit/kafka/utils/TestUtils.scala]: Updated in docs

File: clients/src/test/java/org/apache/kafka/common/security/ssl/SslFactoryTest.java (hunk 2 above)
Comment (abhi-ksolves) [reply clients/src/test/java/org/apache/kafka/common/security/ssl/SslFactoryTest.java]: Reverted

**Comment by abhi-ksolves** [comment]
@chia7712 Mentioned changes are done. Hope this can be merged now.

**Comment by abhi-ksolves** [comment]
Hi @chia7712 
Can you look into this PR

We can merge this?

**Comment by chia7712** [comment]
@abhi-ksolves thanks for this contribution
